crc32.c

```diff
@@ -199,13 +199,78 @@ const z_crc_t FAR * ZEXPORT get_crc_table()
 #define DO8 DO1; DO1; DO1; DO1; DO1; DO1; DO1; DO1
 
 /* ========================================================================= */
-unsigned long ZEXPORT crc32_z(crc, buf, len)
+local
+unsigned long ZEXPORT crc32_table_lookup(crc, buf, len)
     unsigned long crc;
     const unsigned char FAR *buf;
     z_size_t len;
 {
     if (buf == Z_NULL) return 0UL;
 
+    crc = crc ^ 0xffffffffUL;
+    while (len >= 8) {
```

Comment:
This seems to add some considerable amount of code in zlib's crc32 file.

I wonder if you could integrate your patch in a less intrusive way? (i.e. less is better)

I think a good target would be to try to easy the burden of maintenance of zlib.

Reply:
This bit is largely moving the table implementation to its own independent function. It lowers the maintenance burden by uncoupling it from the other optimized implementations.

There is also a bit of boiler plate added here to support static compilation (function pointer), while still providing an optimized glibc implementation.

I could drop out the glibc ifunc however this it a bit that provides an advantage for ARM and PowerPC64 and provides a way to push out the hardware feature detection out of the main line of the crc32 calculation.

I've cleanly separated each commit to make it easier to review though a2f5adc957354d677fe25a7fc167506c436dd6e6 doesn't look as intuitive as the real change is.